### ISSUE TITLE
Lost characters in filenames with `--icons` depending on extension

### ISSUE THREAD

**Issue by aAlexanderLeben** [ogham/exa#976]
# Environment

## The version of exa being used (`exa --version`)

```shell
$ exa --version
exa - list files on the command-line
v0.10.1 [+git] (pre-release debug build!)
https://the.exa.website/
```

## The command-line arguments you are using

- `--long --icons` 
- `--long`

## Your operating system and hardware platform

Tested in Vagrant VM with revision v0.10.1-57-gb32f441

# Bug

When filenames starts with digits and file extension is `mp4` then some characters of the filename are not shown when using `--icons` command line switch. When filename starts with a character, the spacing between icon and filename is wrong.

## How to reproduce

Create a bunch of files with different prefixes and different suffixes:

```shell
for name in 00 00x 10 10x 11 11x 100 100x 1234 1234x a aa aaa; \
do \
  touch ${name}.mp4; \
  touch ${name}.md; \
  touch ${name}; \
done
```
Check filenames with `exa --long` (works as expected) and compare with `exa --long --icons` (broken):
![exa not-working](https://user-images.githubusercontent.com/2428995/139859137-dbab6ebe-cb5d-4e9a-9976-761e06e64c77.png)

**Comment by cyqsimon** [comment]
+1

This issue seems to be platform-dependent. On my Manjaro desktop where I installed `exa` via AUR, this bug seems to be nonexistent; whereas on all my RHEL-based servers (CentOS 7, CentOS 8, RHEL 8) where I installed `exa` via `cargo install` this bug is present. Could also have something to do with SSH but I doubt it.

The consistent way I found to replicate this issue (on platforms where it exists) is as follows:
1. In an empty directory, run `mkdir 1.2.3.4 && mkdir this-should-be-fine`
2. Run `ls` and `exa`, and observe that their outputs are identical, as expected
3. Run `exa --icons`, and observe the bug
![bug-screenshot](https://user-images.githubusercontent.com/28627918/141433686-3d7421ec-7ac5-43a6-94fa-cae9f41da653.png)

**Comment by ariasuni** [comment]
I’m pretty sure it’s some fonts × terminal shenanigans. I’d like to know if this happens on a recent enough distribution, otherwise it’s a probably a bug in some other software that’s already been fixed a while ago.

**Comment by cyqsimon** [comment]
> I’m pretty sure it’s some fonts × terminal shenanigans. I’d like to know if this happens on a recent enough distribution, otherwise it’s probably a bug in some other software that’s already been fixed a while ago.

I understand why you would theorise this but I don't think this is what's happening. I have `exa` installed on both my Manjaro laptop and RHEL8.5 server. Both have the newest version of `exa` and all software updated.

If I run `exa --icons` on my laptop locally, there is no issue. However if I SSH into my server on the same laptop and run `exa --icons` (using the exact same terminal and font settings), the bug occurs.

![different behaviour](https://user-images.githubusercontent.com/28627918/144757953-e945177e-024d-442d-a6d1-2e6554f4a963.png)

**Comment by ariasuni** [comment]
Are you using the same shell in your SSH session and locally? I know that shells sometimes handle characters outside of ASCII very differently (especially here, since Nerd Fonts icons are in private use ranges…).

**Comment by cyqsimon** [comment]
> Are you using the same shell in your SSH session and locally?
Yes, I use bash for everything. It looks different because I've got starship shell prompt installed on my laptop. I can disable it and nothing else changes.

**Comment by aAlexanderLeben** [comment]
I thought it could be related to my setup firstly, that's why I checked with the virtual machine configured by the official Vagrantfile of this project. And issue is visible there as well. Not sure if the VM is to be seen as a reference system, but it looks to me from the outside.
